app/jobs/flood_risk_engine/update_water_management_area_job.rb

```diff
@@ -12,29 +12,29 @@ def perform(location)
     private
 
     def get_area_for(location)
-      api_result = lookup_area_using_using_external_api(location)
+      response = DefraRuby::Area::WaterManagementAreaService.run(location.easting, location.northing)
 
-      if inside_england?(api_result)
-        find_or_create_area_from_api_result(api_result)
-      else
-        WaterManagementArea.outside_england_area
-      end
-    end
+      return process_successful_response(response.area) if response.successful?
 
-    def lookup_area_using_using_external_api(location)
-      coords = EA::AreaLookup::Coordinates.new(easting: location.easting,
-                                               northing: location.northing)
-      EA::AreaLookup.find_water_management_area_by_coordinates(coords)
+      process_unsucessful_response(response)
     end
 
-    def find_or_create_area_from_api_result(api_result)
-      WaterManagementArea.find_or_create_by(code: api_result[:code]) do |area|
-        area.update_attributes(api_result)
+    def process_successful_response(result)
+      WaterManagementArea.find_or_create_by(code: result.code) do |area|
```

Comment:
Have you heard about https://medium.com/aviabird/ruby-tap-that-method-90c8a801fd6a before? feels like it can be useful here :)

Reply:
That's a bit of ruby I never knew about 😁 🤯

However I'm not sure I've got how you think it can help here. I've tested locally with this

```ruby
WaterManagementArea.find_or_create_by(code: result.code).tap do |area|
  area.update_attributes(
    area_id: result.area_id,
    area_name: result.area_name,
    short_name: result.short_name,
    long_name: result.long_name
  )
end
```

But that has just added `.tap` into some code that previously didn't need it. Would be grateful for some advice on what I should be doing.

Reply:
Is quite recent stuff :) And yes, you are right, in here is actually not useful as `find_or_create_by` behaves like tap already.